File: interface.py
from flask import Flask, render_template, jsonify
import redis

app = Flask(__name__)

r = redis.Redis(
    host="localhost",
    port=6379,
    db=0,
    decode_responses=True
)

@app.route("/")
def index():
    return render_template("index.html")

@app.route("/api/chats")
def listar_chats():
    chats = sorted(r.smembers("chats_ativos"))
    return jsonify(chats)

@app.route("/api/historico/<numero>")
def historico(numero):
    msgs = r.lrange(numero, 0, -1)
    return jsonify(msgs)

@app.route("/api/clear/<numero>", methods=["POST"])
def limpar_chat(numero):
    r.delete(numero)
    r.srem("chats_ativos", numero)
    return jsonify({"ok": True})

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8000, debug=False)

File: interface.js
import express from "express";
import { createClient } from "redis";

const app = express();
app.use(express.json());

// Redis
const redis = createClient({
  url: "redis://localhost:6379"
});

redis.connect();

// ========================
// LISTAR CHATS ATIVOS
// ========================
app.get("/api/chats", async (req, res) => {
  const chats = await redis.sMembers("chats_ativos");
  res.json(chats);
});

// ========================
// HISTÓRICO DE UM CHAT
// ========================
app.get("/api/historico/:numero", async (req, res) => {
  const { numero } = req.params;

  const msgIds = await redis.lRange(numero, 0, -1);

  const mensagens = [];
  for (const id of msgIds) {
    const msg = await redis.hGetAll(`msg:${id}`);
    mensagens.push(msg);
  }

  res.json(mensagens);
});

// ========================
app.listen(3000, () => {
  console.log("Servidor rodando em http://localhost:3000");
});
